### PR TITLE
Morph addressbook to entrybook

### PR DESCRIPTION
**Pull request by epicfailname**
Morphs application and features to operate on entries for links and feeds, instead of persons and addresses

* ```Person``` -> ```Entry```
* ```AddressBook``` -> ```EntryBook```
* ```Name``` -> ```Title```
* ```Phone``` -> ```Comment```
* ```Email``` -> ```Link```
* ```Address``` (made invisible, unused field. Kept to pass regression tests, and as a placeholder for future properties of ```Entry```)
* Minimal changes made to reflect the morphed state on the user-side
* Several internal functions and names retain their old names

### CODE REVIEW

**Comment by epicfailname** [comment]
Resolves #30 
Resolves part of #19

**Review by thomastanck — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/EntryCard.java [hunk 1]
```diff
@@ -24,35 +24,36 @@
...
         id.setText(displayedIndex + ". ");
-        name.setText(person.getName().fullName);
-        phone.setText(person.getPhone().value);
-        address.setText(person.getAddress().value);
-        email.setText(person.getEmail().value);
-        person.getTags().forEach(tag -> {
+        title.setText(entry.getTitle().fullTitle);
+        comment.setText(entry.getComment().value);
+        address.setText(entry.getAddress().value);
+        address.setManaged(false);
```
Comment (rlrh) [inline src/main/java/seedu/address/ui/EntryCard.java]: What does this line do? Perhaps we can add a comment about it here for future reference.

**Review by rlrh — DISMISSED** [review]
Looks good, just that it'd be better to add some comments on important changes for future reference.

File: src/main/java/seedu/address/ui/EntryCard.java (hunk 1 above)
Comment (rlrh) [inline src/main/java/seedu/address/ui/EntryCard.java]: What does this line do? I think we should add a comment about it here for future reference.

File: src/main/resources/view/MainWindow.fxml [hunk 2]
```diff
@@ -12,7 +12,7 @@
 <?import javafx.scene.layout.VBox?>
 
 <fx:root type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1"
-         title="Address App" minWidth="450" minHeight="600" onCloseRequest="#handleExit">
+         title="Entrybook App" minWidth="450" minHeight="600" onCloseRequest="#handleExit">
```
Comment (rlrh) [inline src/main/resources/view/MainWindow.fxml]: Small detail but since it's user facing, should we rename this to "Readme App" or just "Readme" instead? Or we can do it in another PR once we have settled on a name.

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 3]
```diff
@@ -31,22 +31,24 @@
...
 
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        Title title = ParserUtil.parseTitle(argMultimap.getValue(PREFIX_TITLE).get());
+        Comment comment = ParserUtil.parseComment(argMultimap.getValue(PREFIX_COMMENT).get());
+        Link link = ParserUtil.parseLink(argMultimap.getValue(PREFIX_LINK).get());
+        // Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS));
```
Comment (rlrh) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java]: So as not to let the changes propagate too far, instead of modifying `ParserUtil`, we can process a missing address argument right here using `Optional.orElse`:
`Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).orElse("Default address"));`
So `ParserUtil#parseAddress` can still retain the same signature as others - `Address parseAddress(String address)` instead of `Address parseAddress(Optional<String> maybeAddress)`

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 4]
```diff
@@ -43,17 +43,17 @@ public EditCommand parse(String args) throws ParseException {
...
+            editPersonDescriptor.setComment(ParserUtil.parseComment(argMultimap.getValue(PREFIX_COMMENT).get()));
         }
-        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
-            editPersonDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+        if (argMultimap.getValue(PREFIX_LINK).isPresent()) {
+            editPersonDescriptor.setLink(ParserUtil.parseLink(argMultimap.getValue(PREFIX_LINK).get()));
         }
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
+            editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS)));
```
Comment (rlrh) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java]: If you take my suggestion to handle a missing address argument in `AddCommandParser`, then this line can go back to the original.

**Review by rlrh — CHANGES_REQUESTED** [review]
Looks good. But I think `ParserUtil#parseAddress` should not be modified, instead `AddCommandParser` should handle a missing address argument through `Optional.orElse`, and another benefit is that `EditCommandParser` will not need to be modified too.
Also, do you want to mark this PR as resolving part of #19?

File: src/main/java/seedu/address/ui/EntryCard.java (hunk 1 above)
Comment (epicfailname) [reply src/main/java/seedu/address/ui/EntryCard.java]: Makes the Address label invisible, graphics pipeline will not manage it. And agreed, will do

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 3 above)
Comment (epicfailname) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java]: Good idea!

**Review by rlrh — APPROVED** [review]
